torch/quantization/_numeric_suite.py

```diff
@@ -39,3 +42,139 @@ def compare_weights(float_dict, quantized_dict):
             weight_dict[key]["float"] = float_dict[match_key]
             weight_dict[key]["quantized"] = quantized_dict[key]
     return weight_dict
+
+
+def _get_observer_dict(mod, target_dict, observer_name, prefix=""):
```

Comment:
Suggest making this a top level API, i.e remove the _ Also, instead of observer name, lets go with observer type.